### ISSUE TITLE
sm_updatemappool "Not enough space on the heap" exception

### ISSUE THREAD

**Issue by walliski** [KZGlobalTeam/gokz#391]
At some point the past month, or maybe even earlier, we have started seeing the exception below when trying to update the ranked map pool. 

GOKZ 3.4.0

```
L 10/20/2022 - 09:46:34: [SM] Exception reported: Not enough space on the heap
L 10/20/2022 - 09:46:34: [SM] Blaming: gokz-localranks.smx
L 10/20/2022 - 09:46:34: [SM] Call stack trace:
L 10/20/2022 - 09:46:34: [SM]   [1] Line 55, gokz-localranks/db/update_ranked_map_pool.sp::DB_UpdateRankedMapPool
L 10/20/2022 - 09:46:34: [SM]   [2] Line 388, gokz-localranks/commands.sp::CommandUpdateMapPool
```

**Comment by walliski** [comment]
I thought this was related to the ever growing map list, but it seems like it might be something else that has broken it.

If there are changes made to how it loads the map, also check #180 as its slightly related.

**Comment by Sikarii** [comment]
I'm surprised this hasn't broken sooner, the compiler reserves 16384 bytes for the heap on 1.10 and 17900 bytes on 1.11:
```
SourcePawn Compiler 1.11.0.6911
Copyright (c) 1997-2006 ITB CompuPhase
Copyright (c) 2004-2021 AlliedModders LLC

Code size:         138652 bytes
Data size:         615960 bytes
Stack/heap size:      17900 bytes
Total requirements:  772512 bytes
```

[The line allocating the heap](https://github.com/KZGlobalTeam/gokz/blob/928b433df27f1d89f8c1383e516d5d32f9278d9b/addons/sourcemod/scripting/gokz-localranks/db/update_ranked_map_pool.sp#L55) allocates 892 times 40 bytes assuming all global maps are in the map pool, that would total to 35680 bytes.

That being said, the fix to this is either to rewrite the logic for pushing the maps, perhaps using transactions, or raising the heap size with the compiler directive `#pragma dynamic <bytes>`.

**Comment by zealain** [comment]
Fixed in 3.5.2
